### PR TITLE
moved labels to the same device as logits for OTP, CODEGEN ,gptj and pixel2struct model

### PR DESCRIPTION
**Pull request by sushmanthreddy**
# What does this PR do?

As suggested in the [#22561](https://github.com/huggingface/transformers/issues/22561), moved labels to the same device as logits for the OTP model and codegen model

@sgugger  can u pls review and merge this pr??

@sgugger I am really sorry for the mess of this pr, I will not repeat this in the future .... pls once review and merge this...I should have created multiple branches for each pr but sorry this will not be repeated...

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for working on this!

**Comment by sushmanthreddy** [comment]
Glad to help!
